### PR TITLE
Add check_if_service_has_usage_on_destroy field to google_project_service resource

### PR DESCRIPTION
**Pull request by shahmedhaCorp**
release-note: b/331417946
Add new flag "check_if_service_has_usage_on_destroy" to resource "google_project_service" to support customer feedback https://github.com/hashicorp/terraform-provider-google/issues/17686

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
resourcemanager: added `check_if_service_has_usage_on_destroy` field to `google_project_service` resource
```

### CODE REVIEW

**Review by NickElliot — CHANGES_REQUESTED** [review]
can you run go fmt on the handwrittenfiles? thanks

**Comment by shahmedhaCorp** [comment]
I only want to make changes to `google-beta` is this the right repo? should i use https://github.com/hashicorp/terraform-provider-google-beta

**Comment by shahmedhaCorp** [comment]
I ran 'make test', 'make lint' and 'make fmt'



```

shahmedha@shahmedha:~/go/src/github.com/hashicorp/terraform-provider-google-beta$ make test
==> Checking that code complies with gofmt requirements...
go vet
go test  -timeout=30s $(go list -e ./... | grep -v github.com/hashicorp/terraform-provider-google-beta/scripts)
?       github.com/hashicorp/terraform-provider-google-beta     [no test files]
?       github.com/hashicorp/terraform-provider-google-beta/google-beta/acctest [no test files]
?       github.com/hashicorp/terraform-provider-google-beta/google-beta/envvar  [no test files]
?       github.com/hashicorp/terraform-provider-google-beta/google-beta/fwmodels        [no test files]
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/functions       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/fwprovider      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/fwresource      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/fwtransport     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/provider        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/provider/universe       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/accessapproval (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/accesscontextmanager   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/activedirectory        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/alloydb        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/apigateway     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/apigee (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/apikeys        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/appengine      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/apphub (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/artifactregistry       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/assuredworkloads       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/backupdr       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/beyondcorp     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/biglake        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigquery       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigqueryanalyticshub   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigqueryconnection     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigquerydatapolicy     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigquerydatatransfer   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigqueryreservation    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/bigtable       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/billing        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/binaryauthorization    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/blockchainnodeengine   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/certificatemanager     (cached)
?       github.com/hashicorp/terraform-provider-google-beta/google-beta/services/clouddomains   [no test files]
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudasset     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudbuild     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudbuildv2   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/clouddeploy    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudfunctions (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudfunctions2        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudidentity  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudids       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudquotas    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrun       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudscheduler (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudtasks     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/composer       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/compute        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/container      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/containeranalysis      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/containerattached      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/containeraws   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/containerazure (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/corebilling    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/databasemigrationservice       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/datacatalog    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dataflow       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dataform       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/datafusion     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/datalossprevention     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/datapipeline   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dataplex       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dataproc       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dataprocmetastore      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/datastore      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/datastream     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/deploymentmanager      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dialogflow     (cached)
?       github.com/hashicorp/terraform-provider-google-beta/google-beta/services/edgenetwork    [no test files]
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dialogflowcx   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/discoveryengine        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/dns    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/documentai     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/documentaiwarehouse    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/edgecontainer  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/essentialcontacts      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/eventarc       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/filestore      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebase       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebaseappcheck       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebasedatabase       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebaseextensions     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebasehosting        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebaserules  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firebasestorage        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/firestore      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/gkebackup      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/gkehub (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/gkehub2        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/gkeonprem      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/healthcare     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/iam2   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/iambeta        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/iamworkforcepool       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/iap    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/identityplatform       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/integrationconnectors  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/integrations   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/kms    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/logging        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/looker (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/managedkafka   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/memcache       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/migrationcenter        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/mlengine       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/monitoring     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/netapp (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/networkconnectivity    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/networkmanagement      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/networksecurity        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/networkservices        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/notebooks      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/orgpolicy      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/osconfig       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/oslogin        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/parallelstore  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/privateca      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/privilegedaccessmanager        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/publicca       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/pubsub (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/pubsublite     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/recaptchaenterprise    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/redis  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/resourcemanager        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/runtimeconfig  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/secretmanager  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/securesourcemanager    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/securitycenter (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/securitycentermanagement       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/securityposture        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/securityscanner        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/servicedirectory       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/servicemanagement      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/servicenetworking      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/serviceusage   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/sourcerepo     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/spanner        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/sql    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/storage        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/storageinsights        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/storagetransfer        (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/tags   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/tpu    (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/tpuv2  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/vertexai       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/vmwareengine   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/vpcaccess      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/workbench      (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/workflows      (cached)
?       github.com/hashicorp/terraform-provider-google-beta/version     [no test files]
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/workstations   (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/sweeper (cached) [no tests to run]
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgdclresource  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgiamresource  (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/tpgresource     (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/transport       (cached)
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/verify  (cached)
shahmedha@shahmedha:~/go/src/github.com/hashicorp/terraform-provider-google-beta$ make lint
==> Checking that code complies with gofmt requirements...
go vet
shahmedha@shahmedha:~/go/src/github.com/hashicorp/terraform-provider-google-beta$ make fmt
==> Fixing source code with gofmt...
gofmt -w -s ./google-beta


```

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go [hunk 1]
```diff
@@ -197,6 +197,41 @@ func testAccCheckProjectService(t *testing.T, services []string, pid string, exp
...
+			},
+			{
+				Config: testAccProjectService_checkUsage(services, pid, org, "true"),
+			},
+			{
+				ResourceName:            "google_project_service.test",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"disable_on_destroy", "disable_dependent_services"},
+			},
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go:230]:
```suggestion
```
if this isn't being verified, there is no point in doing an import step again.

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go [hunk 2]
```diff
@@ -197,6 +197,41 @@ func testAccCheckProjectService(t *testing.T, services []string, pid string, exp
...
+		},
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go:231]: Add another test step here that attempts to disable the service (by using an empty config) with an `ExpectError: regexp.MustCompile("[whatever is returned from the api here I'm not certain of the error message you're expecting]"),`

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go [hunk 3]
```diff
@@ -323,3 +358,20 @@ resource "google_project_service" "test" {
...
+  org_id     = "%s"
+}
+
+resource "google_project_service" "test" {
+  project = google_project.acceptance.project_id
+  service = "%s"
+
+  check_if_service_has_usage_on_destroy = checkIfServiceHasUsage
+}
+`, pid, pid, org, service)
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go:376]:
```suggestion
func testAccProjectService_checkUsage(service string, pid, org string, checkIfServiceHasUsage string) string {
	return fmt.Sprintf(`
resource "google_project" "acceptance" {
  project_id = "%s"
  name       = "%s"
  org_id     = "%s"
}

resource "google_project_service" "test" {
  project = google_project.acceptance.project_id
  service = "%s"

  check_if_service_has_usage_on_destroy = %s
}
`, pid, pid, org, service, checkIfServiceHasUsage)
```

this needs to use formatted string syntax to get in your new argument

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go [hunk 4]
```diff
@@ -123,6 +123,11 @@ func ResourceGoogleProjectService() *schema.Resource {
 				Optional: true,
 				Default:  true,
 			},
+			"check_if_service_has_usage_on_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go:130]:
```suggestion
<% unless version == 'ga' -%>
			"check_if_service_has_usage_on_destroy": {
				Type:     schema.TypeBool,
				Optional: true,
				Default:  false,
			},
<% end -%>
```
example

**Review by NickElliot — CHANGES_REQUESTED** [review]
To make this change only to google-beta, a version guard needs to be added to the new lines of code (example change provided), and both files will need to be changed from `.go` files to `.go.erb` files. See the handwritten sections of https://googlecloudplatform.github.io/magic-modules/develop/resource/

Marked several issues with the test.

Let me know if you need any further assistance!

**Comment by shahmedhaCorp** [comment]
> To make this change only to google-beta, a version guard needs to be added to the new lines of code (example change provided), and both files will need to be changed from `.go` files to `.go.erb` files. See the handwritten sections of https://googlecloudplatform.github.io/magic-modules/develop/resource/
> 
> Marked several issues with the test.
> 
> Let me know if you need any further assistance!

I didn't find any `.go.erb` files under services/resources

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go (hunk 2 above)
Comment (shahmedhaCorp) [reply mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go:225]: > Add another test step here that attempts to disable the service (by using an empty config) with an `ExpectError: regexp.MustCompile("[whatever is returned from the api here I'm not certain of the error message you're expecting]"),`

This new flag is optional so with no input for the flag the disable will be successfull.

**Comment by shahmedhaCorp** [comment]
@NickElliot Friendly Ping for review .

**Review by NickElliot — COMMENTED** [review]
Could you update the test to include a step that purposefully triggers the new error? Or is it not possible to do so?

**Comment by shahmedhaCorp** [comment]
> Could you update the test to include a step that purposefully triggers the new error? Or is it not possible to do so?

It is not possible to test this scenario as we can't mock the usage data in the test.

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb [hunk 5]
```diff
@@ -256,7 +264,15 @@ func resourceGoogleProjectServiceDelete(d *schema.ResourceData, meta interface{}
...
 	service := d.Get("service").(string)
 	disableDependencies := d.Get("disable_dependent_services").(bool)
-	if err = disableServiceUsageProjectService(service, project, d, config, disableDependencies); err != nil {
+	<% unless version == 'ga' -%>
+	checkUsage := d.Get("check_if_service_has_usage_on_destroy").(bool)
+	<% end -%>
+	<% if version == "ga" -%>
+	  err = disableServiceUsageProjectService(service, project, d, config, disableDependencies)
+	<% else -%>
+    err = disableServiceUsageProjectService(service, project, d, config, disableDependencies,checkUsage)
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb:273]: can you make the indentation consistent with the above and below lines here

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb [hunk 6]
```diff
@@ -256,7 +264,15 @@ func resourceGoogleProjectServiceDelete(d *schema.ResourceData, meta interface{}
 
 	service := d.Get("service").(string)
 	disableDependencies := d.Get("disable_dependent_services").(bool)
-	if err = disableServiceUsageProjectService(service, project, d, config, disableDependencies); err != nil {
+	<% unless version == 'ga' -%>
+	checkUsage := d.Get("check_if_service_has_usage_on_destroy").(bool)
+	<% end -%>
+	<% if version == "ga" -%>
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb:270]: can these be converted to an `else`

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb [hunk 7]
```diff
@@ -271,7 +287,11 @@ func resourceGoogleProjectServiceUpdate(d *schema.ResourceData, meta interface{}
 }
 
 // Disables a project service.
+<% if version == "ga" -%>
 func disableServiceUsageProjectService(service, project string, d *schema.ResourceData, config *transport_tpg.Config, disableDependentServices bool) error {
+<% else -%>
+func disableServiceUsageProjectService(service, project string, d *schema.ResourceData, config *transport_tpg.Config, disableDependentServices bool, checkUsage bool) error {
+<% end -%>
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb:294]: can these be inverted to be an unless/else?

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go.erb [hunk 8]
```diff
@@ -104,7 +105,11 @@ func TestAccProjectService_disableDependentServices(t *testing.T) {
 				ResourceName:            "google_project_service.test",
 				ImportState:             true,
 				ImportStateVerify:       true,
+			<% if version == "ga" -%>
 				ImportStateVerifyIgnore: []string{"disable_on_destroy"},
+			<% else -%>
+			  ImportStateVerifyIgnore: []string{"disable_on_destroy", "check_if_service_has_usage_on_destroy"},
+			<% end -%>
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go.erb:112]: can these be inverted to the unless/else, also indentation consistency

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go.erb [hunk 9]
```diff
@@ -164,7 +169,11 @@ func TestAccProjectService_renamedService(t *testing.T) {
 				ResourceName:            "google_project_service.test",
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"disable_on_destroy", "disable_dependent_services"},
+				<% if version == "ga" -%>
+          ImportStateVerifyIgnore: []string{"disable_on_destroy", "disable_dependent_services"},
+        <% else -%>
+          ImportStateVerifyIgnore: []string{"disable_on_destroy", "check_if_service_has_usage_on_destroy", "disable_dependent_services"},
+        <% end -%>
```
Comment (NickElliot) [inline mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service_test.go.erb:176]: also invert to unless/else, and indentation consistency

**Review by NickElliot — CHANGES_REQUESTED** [review]
is there anyway you could trigger it through manual testing on a personal project? would like to verify that it triggers when it expected somehow.

also highlighted a bunch of small fixes

**Comment by shahmedhaCorp** [comment]
> is there anyway you could trigger it through manual testing on a personal project? would like to verify that it triggers when it expected somehow.
> 
> also highlighted a bunch of small fixes

I tried it in gcloud . here are the results . https://screenshot.googleplex.com/CHeK4kH536MSayq

File: mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb (hunk 7 above)
Comment (shahmedhaCorp) [reply mmv1/third_party/terraform/services/resourcemanager/resource_google_project_service.go.erb:294]: Done

**Review by NickElliot — COMMENTED** [review]
> > is there anyway you could trigger it through manual testing on a personal project? would like to verify that it triggers when it expected somehow.
> 
> I tried it in gcloud . here are the results.

Could you use that project in your Terraform config to trigger it via trying to disable the service through Terraform?

**Comment by shahmedhaCorp** [comment]
> > > is there anyway you could trigger it through manual testing on a personal project? would like to verify that it triggers when it expected somehow.
> > 
> > 
> > I tried it in gcloud . here are the results.
> 
> Could you use that project in your Terraform config to trigger it via trying to disable the service through Terraform?

How can I test this feature in terraform config without merging it ?

**Review by NickElliot — COMMENTED** [review]
https://googlecloudplatform.github.io/magic-modules/get-started/generate-providers/

**Comment by shahmedhaCorp** [comment]
> check_if_service_has_usage_on_destroy

tested it manually in terraform and here is the results

```
shahmedha@shahmedha:~/my_space/test$ TF_CLI_CONFIG_FILE="$HOME/tf-dev-override.tfrc" terraform apply
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - hashicorp/google in /usr/local/google/home/shahmedha/go/bin
│  - hashicorp/google-beta in /usr/local/google/home/shahmedha/go/bin
│ 
│ The behavior may therefore not match any released version of the provider and applying changes may cause the state to become incompatible with published releases.
╵
google_project_service.google-beta-instance: Refreshing state... [id=836387143453/serviceusage.googleapis.com]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # google_project_service.google-beta-instance will be destroyed
  # (because google_project_service.google-beta-instance is not in configuration)
  - resource "google_project_service" "google-beta-instance" {
      - check_if_service_has_usage_on_destroy = true -> null
      - disable_on_destroy                    = true -> null
      - id                                    = "836387143453/serviceusage.googleapis.com" -> null
      - project                               = "836387143453" -> null
      - service                               = "serviceusage.googleapis.com" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_project_service.google-beta-instance: Destroying... [id=836387143453/serviceusage.googleapis.com]
╷
│ Error: Error when reading or editing Project Service 836387143453/serviceusage.googleapis.com: Error disabling service "serviceusage.googleapis.com" for project "836387143453": googleapi: Error 400: The service serviceusage.googleapis.com has usage in the last 30 days. Please specify check_if_service_has_usage=SKIP if you want to proceed with disabling the service.
│ Help Token: ARZIt86pto79PoueEfQ2vjFyiljmDus1JOOESW5z0vsRWec2Taf-5C-37TF-szSHX1rXEPBg3pqSXnf0U5GIDjuHAVGOVoBLfoJ6uMBwKvNj2K3Y
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.PreconditionFailure",
│     "violations": [
│       {
│         "subject": "?error_code=100002\u0026service_name=serviceusage.googleapis.com",
│         "type": "googleapis.com"
│       }
│     ]
│   },
│   {
│     "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│     "domain": "serviceusage.googleapis.com",
│     "metadata": {
│       "service_name": "serviceusage.googleapis.com"
│     },
│     "reason": "COMMON_SU_SERVICE_HAS_USAGE"
│   },
│   {
│     "@type": "type.googleapis.com/google.rpc.DebugInfo",
│     "detail": "[ORIGINAL ERROR] tenantmanager::100002: com.google.api.tenant.error.TenantManagerException: The service serviceusage.googleapis.com has usage in the last 30 days. Please specify check_if_service_has_usage=SKIP if you want to proceed with disabling the service. [google.rpc.error_details_ext] { code: 9 message: \"The service serviceusage.googleapis.com has usage in the last 30 days. Please specify check_if_service_has_usage=SKIP if you want to proceed with disabling the service.\\nHelp Token: ARZIt86pto79PoueEfQ2vjFyiljmDus1JOOESW5z0vsRWec2Taf-5C-37TF-szSHX1rXEPBg3pqSXnf0U5GIDjuHAVGOVoBLfoJ6uMBwKvNj2K3Y\" details { [type.googleapis.com/google.rpc.PreconditionFailure] { violations { type: \"googleapis.com\" subject: \"?error_code=100002\u0026service_name=serviceusage.googleapis.com\" } } } details { [type.googleapis.com/google.rpc.ErrorInfo] { reason: \"COMMON_SU_SERVICE_HAS_USAGE\" domain: \"serviceusage.googleapis.com\" metadata { key: \"service_name\" value: \"serviceusage.googleapis.com\" } } } } 391798925 { 1: 100002 }"
│   }
│ ]
│ , failedPrecondition
│ 
│ 
╵
```

**Review by NickElliot — APPROVED** [review]
LGTM!
